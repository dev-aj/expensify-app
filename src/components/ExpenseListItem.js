import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import NumberFormat from 'react-number-format';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <div>
            <Link to={`/edit/${id}`}>
                <h3> {description} </h3>
            </Link>
            
            <p>
            <NumberFormat thousandSeparator={true} thousandsGroupStyle="lakh" prefix={'â‚¹'} displayType="text" value={amount} />
            - 
            {moment(createdAt).format("dddd, MMMM Do YYYY")}
            </p>
        </div>
    );
};


export default ExpenseListItem; 