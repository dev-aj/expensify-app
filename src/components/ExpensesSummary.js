import React from 'react';
import { connect } from 'react-redux';
import NumberFormat from 'react-number-format';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpensesSummary = ({ expenseCount, expenseTotal }) =>{
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    return (
        <div>
            <h1>Viewing { expenseCount } { expenseWord } totalling 
                <NumberFormat 
                    thousandSeparator={true} 
                    thousandsGroupStyle="lakh" 
                    prefix={'â‚¹'} 
                    displayType="text" 
                    value={expenseTotal} 
                />
            </h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount : visibleExpenses.length,
        expenseTotal : selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);

