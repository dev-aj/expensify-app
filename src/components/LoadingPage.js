import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <h1>Hold tight..Getting Your Expenses</h1>
        <img className="loader__image" src="/images/loader.gif" />
    </div>
);

export default LoadingPage; 