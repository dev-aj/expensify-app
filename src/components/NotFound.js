import React from 'react';
import { Link } from 'react-router-dom';


const NotFound = ()=> (
    <div>
        OOOOPPSSS!! this is 404 OOOOPPSSS!! <Link to="/">Go Home</Link> 
    </div>
);

export default NotFound;