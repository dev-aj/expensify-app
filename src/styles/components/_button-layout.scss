.button-layout {
    background:$blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-lg;
    font-weight: 400;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;
}

.button-layout--link {
    background: none;
}

.button-layout--secondary {
    @extend .button-layout;
    background: #888888;
}