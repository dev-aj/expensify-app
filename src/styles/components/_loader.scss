.loader {
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
    width: 100vw;
    background-color: rgb(85, 233, 253) 
}

.loader__image {

}