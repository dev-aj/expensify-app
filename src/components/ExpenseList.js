import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

//Here we are also exporting this component as named export which is not connected to redux store
//this is done to test the component with dynamic props

export const ExpenseList = (props) => (           
    <div>
        {
          props.expenses.length === 0 ? (
            <p>No expenses</p>
          )  : (
            props.expenses.map( (expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />;
            })
          )
        }
        
    </div>
);

// const ConnectedExpenseList = connect((state) => {
//     return {
//         expenses : state.expenses
//     };
// })(ExpenseList);

//The above code can be broken down into following parts

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);