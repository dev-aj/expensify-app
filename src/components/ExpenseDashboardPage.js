import React from 'react';

import ConnectedExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from '../components/ExpensesSummary';

const ExpenseDashboardPage = ()=> (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ConnectedExpenseList />
    </div>
);

export default ExpenseDashboardPage; 