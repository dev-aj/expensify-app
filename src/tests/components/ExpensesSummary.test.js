import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test("Should correctly render expenses Summary with 1 expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount = {1} expensesTotal={236} />);
    expect(wrapper).toMatchSnapshot();
});

test("Should correctly render expenses Summary with multiple expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount = {22} expensesTotal={24536} />);
    expect(wrapper).toMatchSnapshot();
});