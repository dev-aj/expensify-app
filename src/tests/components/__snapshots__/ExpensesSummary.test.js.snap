// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses Summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix="₹"
      suffix=""
      thousandSeparator={true}
      thousandsGroupStyle="lakh"
      type="text"
    />
  </h1>
</div>
`;

exports[`Should correctly render expenses Summary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    22
     
    expenses
     totalling
    <NumberFormat
      allowEmptyFormatting={false}
      allowNegative={true}
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      getInputRef={[Function]}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix="₹"
      suffix=""
      thousandSeparator={true}
      thousandsGroupStyle="lakh"
      type="text"
    />
  </h1>
</div>
`;
